api/share/responses.py

```diff
@@ -0,0 +1,37 @@
+from abc import ABC
+
+
+class Response(ABC):
+    pass
+
+
+class ResponseSuccess(Response):
+
+    def __init__(self, value=None):
+        self.value = value
+
+    def __bool__(self):
+        return True
```

Comment:
same as above. Maybe you're just doing this for clarity in contrast to `ResponseFailure`, then a comment is worth adding.

Reply:
see previous comments

Reply:
yep. and I'm skeptical about this approach. A class shouldn't know anything about how another child of its parent class works.

Reply:
Don't understand. `ResponseSuccess` doesn't know anything about `ResponseFailure`

Reply:
No but `ResponseSuccess` sort of knows that `ResponseFailure` evaluates to `False` when `bool(obj)` is called on it. Because otherwise it would not contain code that is not needed (for any practical reasons), since, as we both agree, `bool(obj)` will return `True` for it anyway without any additional methods.

I really don't see how putting a method that does nothing helps. I think probably the only use case for methods that do nothing is a parent abstract class.

Reply:
> since, as we both agree, `bool(obj)` will return `True` for it anyway without any additional methods.

Yes, but it may not be clear for an outside developer (as you like to talk :) )